alyssa_mckee/Django/restful_users/apps/users/views.py

```diff
@@ -0,0 +1,76 @@
+from django.shortcuts import render, redirect, reverse
+from django.contrib import messages
+from .models import User
+
+# Create your views here.
+def index(req):
+	users = User.objects.all()
+	context = {"users":users}
+	return render(req, "users/index.html", context)
+
+def new(req):
+	#add to database
+	context = {}
+	if 'errors'in req.session:
+		context["errors"] = req.session['errors']
+		context["data"]=req.session['create_data']
+		req.session.pop('create_data')
+		req.session.pop('errors')
+	return render(req, "users/new_user.html", context)
+
+def edit(req, id):
+	#if a field empty, fill with current value
+	user = User.objects.filter(id=id)
+	if len(user) == 0: #if user doesn't exist
+		return redirect(reverse("all_users"))
+	user = user[0]
```

Comment:
I would recommend using .first() instead of directly indexing with [0] in django. The .first() method will return the first object in the list if it exists or nothing. Essentially .first() can do a bit of this logic for you.